plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -69,6 +70,10 @@ export class PersonState {
         this.db = db
         this.statsd = statsd
         this.personManager = personManager
+
+        // If set to true, we'll update `is_identified` at the end of `updateProperties`
+        // :KLUDGE: This is an indirect communication channel between `handleIdentifyOrAlias` and `updateProperties`
+        this.updateIsIdentified = false
```

Comment:
Perhaps this could just be returned by `handleIdentifyOrAlias()`, given that `updateProperties()` is only called in `update()`, right after `handleIdentifyOrAlias()`? This being an _indirect communication channel_ makes the update process harder to read.

Reply:
I agree but this became more or less impossible to implement without rebuilding the (finnicky) retry mechanism from scratch. I'm avoiding doing so since that's a risky refactor and I'm not still 100% up-to-date with all the logic involved - it will be costly both from a time- and operational risk standpoint.

Note I'll need to do similar hackery same for person merging/creation within `handleIdentifyOrAlias` as well.

As such I'd rather leave refactoring that huge can of worms to another day and instead make this class stateful. It's called `PersonState` already so not feeling too bad about it, especially with proper unit test coverage to lead future refactorings. :sweat_smile: